6212 Иванов Максим Алексеевич Лаб.3 Вар 26

Gwyn1300 opened the pull request:
**ФИО:** Фамилия Имя  
**Номер группы:** 6212 
**Номер лабораторной:** 3 
**Номер варианта:** 26  
**Краткое описание решаемой задачи:** Согласно варианту 26 лабораторной 3 мне надо было выполнить ускорение аудиофайла в заданное количество раз.  
**Краткое описание работы кода:** Модуль `graph.py` создает и сохраняет график изменения файла. Модуль `process_audiofile.py` обрабатывает аудиофайл. Модуль `main.py` является точкой входа в приложение. Чтобы его запустить, требуется указать аргументы командной строки: `main.py  "Путь к файлу" число, во сколько раз ускорить -s 'Путь к директории, куда сохранить ускоренный аудиофайл' -g 'Путь к директории, куда сохранить график'.

Gwyn1300 committed:
Начало работы над лабой 2

Gwyn1300 committed:
Добавлено скачивание файлов
Добавлено создание .csv файла

Gwyn1300 committed:
Добавление итератора
Разделение кода на функции

Gwyn1300 committed:
Выполнено

Gwyn1300 committed:
Merge branch 'main' into lab2

Gwyn1300 committed:
Исправил

Gwyn1300 committed:
Реализовано увеличение скорости

Gwyn1300 committed:
Реализовано построение графика

Gwyn1300 committed:
Добавлен аргумент пути сохранения нового файла

Gwyn1300 committed:
Выполнена лабораторная работа

IlyaOv commented on lab3/graph.py:
```diff
@@ -0,0 +1,93 @@
+'''
+Файл для работы с графиками
+'''
+import os
+import matplotlib.pyplot as plt
+from matplotlib.figure import Figure
```
Правила оформления импортов: https://peps.python.org/pep-0008/#imports

IlyaOv commented on lab3/graph.py:
```diff
@@ -0,0 +1,93 @@
+import os
+import matplotlib.pyplot as plt
+from matplotlib.figure import Figure
+
```
Две пустые строки

IlyaOv commented on lab3/graph.py:
```diff
@@ -0,0 +1,93 @@
+
+    plt.tight_layout(rect=[0, 0.08, 1, 0.95])
+    return fig
+
```
Две пустые строки: https://peps.python.org/pep-0008/#blank-lines

IlyaOv commented on lab3/process_audiofile.py:
```diff
@@ -0,0 +1,88 @@
+'''
+import soundfile as sf
+import numpy as np
+from graph import create_comparison_charts, save_graph
```
Импорты

IlyaOv commented on lab3/process_audiofile.py:
```diff
@@ -0,0 +1,88 @@
+import soundfile as sf
+import numpy as np
+from graph import create_comparison_charts, save_graph
+
```
Две пустые строки

IlyaOv commented on lab3/process_audiofile.py:
```diff
@@ -0,0 +1,88 @@
+    except OSError as e:
+        print(f"Ошибка чтения файла '{path}': {e}")
+        return None
+
```
Две пустые строки

IlyaOv commented on lab3/process_audiofile.py:
```diff
@@ -0,0 +1,88 @@
+    except OSError as e:
+        print(f"Ошибка при записи файла '{path+"\\audio_comparison_charts.png"}': {e}")
+    return False
+
```
Две пустые строки

IlyaOv commented on lab3/main.py:
```diff
@@ -0,0 +1,25 @@
+Главная функция
+'''
+import argparse
+from process_audiofile import boost_audio
```
Импорты

IlyaOv commented on lab3/main.py:
```diff
@@ -0,0 +1,25 @@
+'''
+import argparse
+from process_audiofile import boost_audio
+
```
Две пустые строки

IlyaOv commented on lab3/process_audiofile.py:
```diff
@@ -0,0 +1,88 @@
+            result[i] = np.mean(arr[start:end], axis=0)
+
+    return result
+
```
Две пустые строки

IlyaOv commented on lab3/process_audiofile.py:
```diff
@@ -0,0 +1,88 @@
+        print(f"Ошибка форматирования данных: {e}")
+    except OSError as e:
+        print(f"Ошибка ввода-вывода: {e}")
+
```
Две пустые строки

IlyaOv reviewed (changes requested):
Замечания, которые относятся к оформлению по прежнему актуальны также для ЛР 1 и 2.

По 3 ЛР:
1. У Вас же вроде по заданию уменьшить скорость надо?
2. Добавьте получившиеся графи в репозиторий.

Gwyn1300 committed:
Исправил форматирование
Добавил функцию уменьшения скорости аудиофайла. Если передавать значение меньше 0, то айдиофайл будет замедляться, если больше, то ускоряться
Добавлен график, при уменьшении скорости в 2 раза
